tincr/io/library/genCellLibrary.tcl

```diff
@@ -339,21 +342,12 @@ proc doPorts { fo } {
     puts $fo "      </bels>"
     puts $fo "    </cell>"
 
-    puts $fo "    <cell>"
-    puts $fo "      <type>OPORT</type>"
-    puts $fo "          <is_port/>"
-    puts $fo "      <level>LEAF</level>"
-    puts $fo "      <pins>"
-    puts $fo "        <pin>"
-    puts $fo "          <name>PAD</name>"
-    puts $fo "          <direction>input</direction>"
-    puts $fo "        </pin>"
-    puts $fo "      </pins>"
-    puts $fo "      <bels>"
+    printPortHeader $fo "OPORT" "input"
     
     dict for {type site} $dict {
-        set b [get_bels -of $site]
-        set ispad 0
+	if {![get_property IS_PAD $site]} {
+	    continue;
+	}
         foreach b [get_bels -of $site] {
             if { [get_property TYPE $b] == "PAD" } {
                 if { [get_property NUM_INPUTS $site] > 0 } {
```

Comment:
One last comment. You should probably move this check outside of the inner for loop (it should be done once per site instead of once per BEL). The same applies for the other loops. Something like: 
```
if {![get_property IS_PAD $site] || [get_property NUM_INPUTS $site] == 0 } {
    continue;
 }
```

You could also just do...
```
if {[get_property IS_PAD $site] && [get_property NUM_INPUTS $site] > 0 } {
    # your code here
 }
```

whichever you prefer